lib/shared/src/models/sync.ts

```diff
@@ -474,7 +487,11 @@ async function fetchServerSideModels(
  * @param {ServerModel[]} models - An array of models from the site config.
  * @returns {ServerModel[]} - The array of models with adjusted context windows where applicable.
  */
-export const maybeAdjustContextWindows = (models: ServerModel[]): ServerModel[] =>
+export const maybeAdjustContextWindows = (
+    models: ServerModel[],
+    isDotComUser: boolean,
+    longContextWindowFlag?: boolean
```

Comment:
```suggestion
    isProUser: boolean,
    isLongContextWindowFlagEnabled?: boolean
```
you are only checking if the model is a free model or not, but you are not checking if the user is Pro user or Free users, you are only checking if they are dot com users. 


The right approach would be only increase the context window for:
1. isProUser
2. isLongContextWindowFlagEnabled